### PR TITLE
Features/i18n locale in filters

### PR DESCRIPTION
**Pull request by mfrachet**
### What does it do?

Allows managing filters from query string when using I18N. The idea is the following:

- At some point, the ContentManager dispatches an action to set some of its layout information in the redux store called `initialParams`. This information is about the pageSize, sorting, etc... **but also about personalization**. A plugin can add keys to this `initialParams`.
- An I18N (redux) middleware catches that action, and modifies the `initialParams` key to add the a `pluginOption.locale` key inside
- What is really cool about the previous point is that the CM knows how to create a query string from a `pluginOption` object. And it can do that **without knowing the shape of the object**. It means that we can decide to add any kind of value in the redux store for the `initialParams` key, and then use that information on the URL for filtering.

This is how we can finally generate URLs like: 

```
?page=1&pageSize=10&_sort=name:ASC&_where[0][name]=hello world&_where[1][i18n][locale]=en
```

### CODE REVIEW

File: packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { useSelector } from 'react-redux';
 import { Picker, Padded, Text, Flex } from '@buffetjs/core';
 import { Carret } from 'strapi-helper-plugin';
 import styled from 'styled-components';
+import get from 'lodash/get';
```
Comment (soupette) [inline packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js:6]: `import { get } from 'lodash';`

**Review by soupette — CHANGES_REQUESTED** [review]
LGTM (after you fix the small requested change)!


Good job!

File: packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js (hunk 1 above)
Comment (mfrachet) [reply packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js:6]: It treeshakes better when imported using import get from 'lodash/get';

File: packages/strapi-plugin-content-manager/admin/src/containers/ListView/utils/buildQueryString.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+   */
+  const { pluginOptions: _, ...otherQueryParams } = {
+    ...queryParams,
+    _where: _where.concat(pluginOptionArray),
+  };
+
+  return `?${stringify(otherQueryParams, { encode: false })}`;
+};
+
+export default buildQueryString;
```
Comment (HichamELBSI) [inline packages/strapi-plugin-content-manager/admin/src/containers/ListView/utils/buildQueryString.js:26]: 👍🏽

File: packages/strapi-plugin-i18n/admin/src/middlewares/tests/localeQueryParamsMiddleware.test.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    middleware(nextFn)(action);
+
+    expect(nextFn).toBeCalledWith(action);
+    expect(action).toEqual({
+      contentType: { pluginOptions: { i18n: { localized: true } } },
+      initialParams: { pluginOptions: { locale: 'en', hello: 'world' } },
+      type: 'ContentManager/ListView/SET_LIST_LAYOUT ',
+    });
+  });
+});
```
Comment (HichamELBSI) [inline packages/strapi-plugin-i18n/admin/src/middlewares/tests/localeQueryParamsMiddleware.test.js:86]: Good job 👍🏽

File: packages/strapi-plugin-content-manager/admin/src/containers/ListView/actions.js [hunk 4]
```diff
@@ -78,6 +78,23 @@ export function toggleModalDelete() {
   };
 }
 
-export const setLayout = layout => ({ layout, type: SET_LIST_LAYOUT });
+export const setLayout = contentType => {
+  const { layouts, settings } = contentType;
+  const defaultSort = `${settings.defaultSortBy}:${settings.defaultSortOrder}`;
```
[the remark this answers is not in the document]
Comment (mfrachet) [reply packages/strapi-plugin-content-manager/admin/src/containers/ListView/actions.js:83]: I've taken this for granted, and I think yes. Can you confirm @soupette ?

[the remark this answers is not in the document]
Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/containers/ListView/actions.js:83]: Yes they are always defined, it's a setting

File: packages/strapi-plugin-content-manager/admin/src/containers/ListView/ListViewLayout.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import PropTypes from 'prop-types';
+import { setLayout } from './actions';
+import ListView from '.';
+import { useQueryParams } from '../../hooks';
+
+const ListViewLayout = ({ layout, ...props }) => {
+  const dispatch = useDispatch();
+  const initialParams = useSelector(state => state.get('content-manager_listView').initialParams);
+  // eslint-disable-next-line no-unused-vars
+  const [_, setQuery] = useQueryParams(initialParams);
```
Comment (soupette) [inline packages/strapi-plugin-content-manager/admin/src/containers/ListView/ListViewLayout.js:12]: `const [, setQuery] = useQueryParams(initialParams);`

So eslint does not complain

**Review by soupette — APPROVED** [review]
LGTM

File: packages/strapi-plugin-content-manager/admin/src/containers/ListView/ListViewLayout.js (hunk 5 above)
Comment (mfrachet) [inline packages/strapi-plugin-content-manager/admin/src/containers/ListView/ListViewLayout.js:12]:
```suggestion
  const [, setQuery] = useQueryParams(initialParams);
```

**Pull request merged**